Python/wolfssl-python-3.8.5.patch

```diff
@@ -0,0 +1,1995 @@
+         self.assertTrue(stats['session_reused'])
+         session2 = stats['session']
+         self.assertEqual(session2.id, session.id)
+@@ -4297,8 +4561,10 @@
+         self.assertNotEqual(session3.id, session.id)
+         self.assertNotEqual(session3, session)
+         sess_stat = server_context.session_stats()
+-        self.assertEqual(sess_stat['accept'], 3)
+-        self.assertEqual(sess_stat['hits'], 1)
++        if not ssl.IS_WOLFSSL:
++            #skip sub-test for sess_accept and sess_hits because those API are STUB
++            self.assertEqual(sess_stat['accept'], 3)
++            self.assertEqual(sess_stat['hits'], 1)
+ 
+         # reuse session again
+         stats = server_params_test(client_context, server_context,
+@@ -4310,8 +4576,10 @@
+         self.assertGreaterEqual(session4.time, session.time)
+         self.assertGreaterEqual(session4.timeout, session.timeout)
+         sess_stat = server_context.session_stats()
+-        self.assertEqual(sess_stat['accept'], 4)
+-        self.assertEqual(sess_stat['hits'], 2)
++        if not ssl.IS_WOLFSSL:
++            #skip sub-test for sess_accept and sess_hits because those API are STUB
++            self.assertEqual(sess_stat['accept'], 4)
++            self.assertEqual(sess_stat['hits'], 2)
+ 
+     def test_session_handling(self):
+         client_context, server_context, hostname = testing_context()
+@@ -4497,10 +4765,17 @@
+             with client_context.wrap_socket(socket.socket(),
+                                             server_hostname=hostname) as s:
+                 s.connect((HOST, server.port))
+-                with self.assertRaisesRegex(ssl.SSLError, 'not server'):
+-                    s.verify_client_post_handshake()
+-                s.write(b'PHA')
+-                self.assertIn(b'extension not received', s.recv(1024))
++                # Error strings returned from wolfSSL are worded differently
++                if ssl.IS_WOLFSSL:
++                    with self.assertRaisesRegex(ssl.SSLError, 'wrong client/server type'):
++                        s.verify_client_post_handshake()
++                    s.write(b'PHA')
++                    self.assertIn(b'Client will not do post handshake authentication', s.recv(1024))
++                else:
++                    with self.assertRaisesRegex(ssl.SSLError, 'not server'):
++                        s.verify_client_post_handshake()
++                    s.write(b'PHA')
++                    self.assertIn(b'extension not received', s.recv(1024))
+ 
+     def test_pha_no_pha_server(self):
+         # server doesn't have PHA enabled, cert is requested in handshake
+@@ -4616,7 +4891,11 @@
+                                             server_hostname=hostname) as s:
+                 s.connect((HOST, server.port))
+         # header, 5 lines for TLS 1.3
+-        self.assertEqual(self.keylog_lines(), 6)
++        if ssl.IS_WOLFSSL:
++            # Does not include EXPORTER
```

Comment:
What does this mean? There's a function we don't support? Can these comments be more specific about what that function is?